src/System.Private.Uri/src/System/Uri.cs

```diff
@@ -4672,6 +4672,11 @@ private unsafe Check CheckCanonical(char* str, ref ushort idx, ushort end, char
                         res |= Check.NotIriCanonical;
                     }
                 }
+                else if ((c >= '{' && c <= '}'))
```

Comment:
I assume doing a range check like this is also including the "|" character?  If so, it would be helpful to add a comment here.  The Uri code is already terribly complicated and anything we can do to improve its readability is useful.

Otherwise, I keep having to look at the ASCII chart: 
![image](https://user-images.githubusercontent.com/7684835/56426818-6613a980-626e-11e9-828f-34b253cbd965.png)

```c#
else if ((c >= '{' && c <= '}')) // includes '{', '|', '}'
```